### PR TITLE
Rename polymorphic function and function

### PR DESCRIPTION
**Pull request by ageron**
**TL;DR**: this PR renames:
* `Function` to `ConcreteFunction`
* `PolymorphicFunction` to `Function`

**Motivation**

`PolymorphicFunction` is a central part of TensorFlow 2.0, so it should have a simple and intuitive name. The fact that it's polymorphic is somewhat of a technical detail. This makes it harder to casually reference this object. Notice for example that the documentation of `tf.function` never mentions this name, and instead it resorts to saying `"the callable returned by function"`. 

This PR renames `PolymorphicFunction` to `Function`, so people can simply refer to them as "tensorflow functions" or "tf functions" (I suggest to avoid calling them simply "functions", as in many contexts this is ambiguous).
Since the name `Function` is already taken by the concrete functions generated by `get_concrete_function()`, this PR renames this class to `ConcreteFunction`, so people can just call them "concrete functions" or "concrete TF functions" (I suggest to avoid calling them "graph functions", as it is unclear whether this refers to TF functions or concrete functions).

**Alternative**

I noticed that concrete functions are often named "graph functions" in many of the comments and variable/function names. This name is pretty natural and explicit, and it would avoid ambiguity when talking about "functions" (are we talking about Python functions? TF Functions? Concrete Functions?). Perhaps it might be a better idea to use `GraphFunction` instead of `Function`, and `ConcreteGraphFunction` instead of `ConcreteFunction`. I'm happy to update this PR if you feel this naming would be preferable.

### CODE REVIEW

**Review by alextp — DISMISSED** [review]
Thanks!

**Comment by rthadur** [comment]
@alextp  seems some checks are failing , can you please verify once

**Comment by alextp** [comment]
There's an "assertRaises" test in function_test.py:1734 which tests for the name ConcreteFunction. Can you change the name in the regexp?

(I _hate_ those tests...)

**Comment by rthadur** [comment]
@ageron can you please make appropriate changes

**Comment by ageron** [comment]
I just merged with HEAD, resolved some conflicts and fixed the issue. You can merge when you want if the tests pass, or you can wait for me to also rename the Polymorphic/Monomorphic protos and `SavedPolymorphicFunction`, as you prefer.

**Review by alextp — DISMISSED** [review]

**Comment by alextp** [comment]
Can you resolve the conflicts?

**Review by allenlavoie — DISMISSED** [review]

**Comment by rthadur** [comment]
@ageron can you resolve conflicts ?

**Review by allenlavoie — DISMISSED** [review]

**Review by allenlavoie — DISMISSED** [review]

**Comment by ageron** [comment]
Yikes, I had to do another merge with conflicts, Polymorphic keeps popping back up, it's like whack a mole!  ;-)

**Review by allenlavoie — DISMISSED** [review]
If merging gets to be too much, consider committing just a change to the user-facing classes to start (in function.py/def_function.py).

**Comment by ageron** [comment]
Thanks @allenlavoie , indeed, I just resolved conflicts again.  If it fails once more, I'll just focus on the user facing classes.

**Review by allenlavoie — DISMISSED** [review]

**Review by allenlavoie — APPROVED** [review]

**Review by alextp — APPROVED** [review]

**Comment by allenlavoie** [comment]
Great names. Thanks for dealing with all of the merge conflicts.

**Comment by ageron** [comment]
Cool, thanks for your help @allenlavoie and @alextp !
